File: frontend/src/lib/types/analytics.ts
export interface IntegrationPlaceholder {
  status: string;
  label: string;
}

export interface AnalyticsOverview {
  total_meetings: number;
  meetings_this_week_previous: number;
  meetings_this_week_upcoming: number;
  recorded_hours: number;
  integrations: IntegrationPlaceholder;
}

File: src/models/analytics.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct IntegrationPlaceholder {
    pub status: String,
    pub label: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AnalyticsOverview {
    pub total_meetings: i64,
    pub meetings_this_week_previous: i64,
    pub meetings_this_week_upcoming: i64,
    pub recorded_hours: f64,
    pub integrations: IntegrationPlaceholder,
}
